core/templates/pages/exploration-editor-page/changes-in-human-readable-form/changes-in-human-readable-form.component.html

```diff
@@ -147,8 +103,136 @@
+                  {{ lostChange.newValue.correct_answer }}
+                </div>
+              </div>
+              <div class="sub-edit">
+                <i>Explanation: </i>
+                <div class="explanation" [innerHTML]="lostChange.newValue.explanation.html">
+                </div>
+              </div>
+            </div>
+
+            <div *ngIf="lostChange.isNewValueEmpty()" class="state-edit-desc">
+              <span>Removed solution</span>
+            </div>
+          </div>
+
         </ng-container>
       </div>
+      <div *ngIf="lostChange.cmd === 'edit_exploration_property'">
+        <ng-container [ngSwitch]="lostChange.propertyName">
+          <div *ngSwitchCase="'title'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New title: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed title</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'category'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New category: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed category</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'objective'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New objective: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed objective</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'language_code'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New language code: </strong >{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed language code</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'init_state_name'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New initiating state name: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed initiating state name</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'tags'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New tags: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed tags</strong>
+            </div>
+          </div>
+        </ng-container>
+      </div>
+      <div *ngIf="lostChange.cmd === 'add_written_translation'">
+        <strong>Change in translation</strong>
+        <div class="sub-edit">
+          <div class="language_code">
+            <i>Language Code: </i> {{lostChange.languageCode}}
```

Comment:
Users may not know the associated language given the language code. Should we map the language name and show it?

Reply:
Done.